scripts/quests/windurst/One_Good_Deed.lua

```diff
@@ -0,0 +1,143 @@
+-----------------------------------
+-- Orastery Woes
+-- Kuroido-Moido !pos -112.5 -4.2 102.9 240
+-- qm1 !pos 197 -8 -27.5 122
+-----------------------------------
+require('scripts/globals/interaction/quest')
+require('scripts/globals/keyitems')
+require('scripts/globals/npc_util')
+require('scripts/globals/quests')
+require('scripts/globals/status')
+require('scripts/globals/items')
+-----------------------------------
+local bibikiID = require('scripts/zones/Bibiki_Bay/IDs')
+-----------------------------------
+
+local quest = Quest:new(xi.quest.log_id.WINDURST, xi.quest.id.windurst.ONE_GOOD_DEED)
+
+quest.reward =
+{
+    gil = 3200,
+    fame = 50,
+    fameArea = xi.quest.fame_area.WINDURST,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+            player:getFameLevel(xi.quest.fame_area.WINDURST) >= 6
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Chipmy-Popmy'] = quest:progressEvent(594),
+
+            onEventFinish =
+            {
+                [594] = function(player, csid, option, npc)
+                    if option == 1 then
+                        quest:begin(player)
+                    end
+                end,
+            },
+        }
+    },
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Chipmy-Popmy'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 2 and player:hasKeyItem(xi.ki.DEED_TO_PURGONORGO_ISLE) then
+                        return quest:progressEvent(595, 0, 614)
+                    elseif quest:getVar(player, 'Prog') == 4 then
+                        return quest:progressEvent(597) -- Complete Quest
+                    end
+                end,
+            },
+            onEventFinish =
+            {
+                [595] = function(player, csid, option, npc)
+                        quest:setVar(player, 'Prog', 3)
+                end,
+                [597] = function(player, csid, option, npc)
+                    quest:complete(player)
+            end,
+            },
+        },
+        [xi.zone.BIBIKI_BAY] =
+        {
+            ['qm_deed'] =
+            {
+                onTrigger = function(player, npc)
+                    local check = true
```

Comment:
Could this all be handled with npcUtil.popFromQM?

Reply:
Never heard of this

Reply:
This a global util function that checks if mobs are spawned or not and if not will spawn, and has params for other things  can see it here.
https://github.com/LandSandBoat/server/blob/fb6f978310c1dd0538f54aca6ae4a54542688283/scripts/globals/npc_util.lua#L36

Reply:
Thanks for pointing this out to me! I'll look into it